src/lightning/fabric/connector.py

```diff
@@ -468,8 +471,8 @@ def _check_and_init_precision(self) -> Precision:
             device = "cpu" if self._accelerator_flag == "cpu" else "cuda"
 
             if isinstance(self.strategy, FSDPStrategy):
-                return FSDPPrecision(precision=self._precision_input, device=device)
-            return MixedPrecision(precision=self._precision_input, device=device)
+                return FSDPPrecision(precision=self._precision_input, device=device)  # type: ignore[arg-type]
+            return MixedPrecision(precision=self._precision_input, device=device)  # type: ignore[arg-type]
```

Comment:
weird mypy thinks we are passing in "16-true" even though above we check `self._precision_input in ("16-mixed", "bf16-mixed")`

Reply:
Probably a bug in mypy